pkg/chunk/local/fs_object_client.go

```diff
@@ -127,12 +139,26 @@ func (f *FSObjectClient) List(ctx context.Context, prefix string) ([]chunk.Stora
 }
 
 func (f *FSObjectClient) DeleteObject(ctx context.Context, objectKey string) error {
-	err := os.Remove(path.Join(f.cfg.Directory, objectKey))
-	if err != nil && os.IsNotExist(err) {
-		return chunk.ErrStorageObjectNotFound
+	// inspired from https://github.com/thanos-io/thanos/blob/55cb8ca38b3539381dc6a781e637df15c694e50a/pkg/objstore/filesystem/filesystem.go#L195
+	file := filepath.Join(f.cfg.Directory, objectKey)
+
+	for file != f.cfg.Directory {
+		if err := os.RemoveAll(file); err != nil {
```

Comment:
We want `os.Remove` only, not `All`, as that will clean non-empty directories too. It will also remove entire sub-tree, if passed objectKey points to a directory!